src/Marten.Testing/Linq/query_with_inner_query.cs

```diff
@@ -0,0 +1,119 @@
+                    IReadonlyCollectionOfInnerClasses = new List<TypeWithInnerCollections>
+                    {
+                        new TypeWithInnerCollections()
+                        {
+                            Id = Guid.NewGuid(),
+                            Array = array,
+                            Flatten = array.Aggregate((i, j) => i + j),
+                            Enumerable = array.AsEnumerable(),
+                            IEnumerableFromArray = array,
+                            IEnumerbaleFromList = array.ToList(),
+                            List = array.ToList(),
+                            IList = array.ToList(),
+                            IReadonlyCollection = array.ToList(),
+                        }
+                    }
+                };
+            }
+        }
+
+        private static readonly TypeWithInnerCollections[] TestData = new TypeWithInnerCollections[]
+        {
+            TypeWithInnerCollections.Create("one", "two"),
+            TypeWithInnerCollections.Create("two", "three"),
+            TypeWithInnerCollections.Create("four", "five"),
+        };
+
+        private const string SearchPhrase = "two";
+
+        public static readonly TheoryData<Expression<Func<TypeWithInnerCollections, bool>>> Predicates = new TheoryData<Expression<Func<TypeWithInnerCollections, bool>>>
+        {
+            x => x.Array.Contains(SearchPhrase),
+            x => x.Enumerable.Contains(SearchPhrase),
+            x => x.IEnumerableFromArray.Contains(SearchPhrase),
+            x => x.IEnumerbaleFromList.Contains(SearchPhrase),
+            x => x.List.Contains(SearchPhrase),
+            x => x.IList.Contains(SearchPhrase),
+            x => x.IReadonlyCollection.Contains(SearchPhrase),
+            x => x.IReadonlyCollection.Where(e => e == SearchPhrase).Any(),
+            x => x.IReadonlyCollectionOfInnerClasses.Where(e => e.Flatten == "onetwo").Any() || x.IReadonlyCollectionOfInnerClasses.Where(e => e.Flatten == "twothree").Any()
+        };
+
+        [Theory]
+        [MemberData(nameof(Predicates))]
+        public async Task can_query_against_array_of_string(Expression<Func<TypeWithInnerCollections, bool>> predicate)
+        {
+            ConfigureStoreToHaveJsonTypeNameHandlingNone();
+            SetupTestData();
+
+            using (var query = theStore.QuerySession())
+            {
+                var results = await query.Query<TypeWithInnerCollections>()
+                    .Where(predicate)
+                    .ToListAsync();
+
+                results.Count.ShouldBe(2);
+                results.All(e => e.Enumerable.Contains(SearchPhrase)).ShouldBeTrue();
+            }
+        }
+
+        private void ConfigureStoreToHaveJsonTypeNameHandlingNone()
+        {
+            var serializer = new JsonNetSerializer();
+            serializer.Customize(s =>
+            {
+                s.TypeNameHandling = Newtonsoft.Json.TypeNameHandling.None;

```

Comment:
I believe the same can be applied as an attribute on the IEnumerable properties on the class to achieve the same rather than customizing the serializer at the store level. Thoughts?

Reply:
@mysticmind good catch, I'll add next case checking that